### PR TITLE
Added support for metadata in messages

### PR DESCRIPTION
**Pull request by angiurgiu**
Added support for metadata in messages
Updated Swagger to version 2021-04-05-preview6 for Metadata support
Updated tests to cover the new property in Send and Update Message operations
Updated all Session Records to reflect the version change to 2021-04-05-preview6

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/UpdateChatMessageOptions.java [hunk 1]
```diff
@@ -35,4 +42,24 @@ public UpdateChatMessageOptions setContent(String content) {
...
+     * Get the metadata property: Message metadata.
+     *
+     * @return the metadata value.
+     */
+    public Map<String, String> getMetadata() {
+        return this.metadata;
+    }
+
+    /**
+     * Set the metadata property: Message metadata.
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/UpdateChatMessageOptions.java:56]: Maybe explain the consequence here. What will happen to the metadata? Replace existing set with the new set? Or Update the one with same name but append new ones? Will old ones be removed?

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadClientTest.java [hunk 2]
```diff
@@ -361,6 +364,7 @@ public void canUpdateExistingMessageWithResponse(HttpClient httpClient) {
 
         ChatMessage message = chatThreadClient.getMessage(response.getId());
         assertEquals(message.getContent().getMessage(), updateMessageRequest.getContent());
+        assertTrue(message.getMetadata().equals(updateMessageRequest.getMetadata()));
     }
 
     @ParameterizedTest
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadClientTest.java:370]: Maybe add new test case to update metadata to prove expected behavior. New ones are added. Same named ones updated. Old ones removed if that's what expected to happen

File: sdk/communication/azure-communication-chat/swagger/README.md [hunk 3]
```diff
@@ -21,18 +21,18 @@ There is one swagger for Chat management APIs.
...
 To update generated files for chat service, run the following command
 
-> autorest README.md --java --v4 --use=@autorest/java@4.0.11 --use=@autorest/modelerfour@4.15.442
+> autorest README.md --java --v4 --use=@autorest/java@4.0.20 --use=@autorest/modelerfour@4.15.442
 
 ### Code generation settings
 ``` yaml
-tag: package-chat-2021-03-07
-require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/896d05e37dbb00712726620b8d679cc3c3be09fb/specification/communication/data-plane/Chat/readme.md
+input-file:
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-chat/swagger/README.md:34]: Is it possible to still follow the existing convention? We are supposed to update the rest-api repo first. If a new commit happened already, why wasn't the readme.md update in rest-api repo?

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/UpdateChatMessageOptions.java (hunk 1 above)
Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/UpdateChatMessageOptions.java:56]: This comment/class is coming directly from the REST Swagger and it's the same between all languages in all of our SDKs. The other SDKs and our API also don't clarify whether it overwrites the entire dictionary or just the keys in the update metadata. But I can log a bug on the API to consider updating the comment to be more clear

File: sdk/communication/azure-communication-chat/swagger/README.md (hunk 3 above)
Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/swagger/README.md:34]: I don't think so, the new swagger from the rest-api specs (**https://raw.githubusercontent.com/Azure/azure-rest-api-specs/08ef4099dff05de3ce1682d58cd4e0e2b1565905/specification/communication/data-plane/Microsoft.CommunicationServicesChat/preview/2021-04-05-preview6/communicationserviceschat.json**) doesn't seem compatible with that convention anymore. 

C# also made this convention change to update to the 2021-04-05-preview6 version: https://github.com/Azure/azure-sdk-for-net/pull/21097/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-854b8aa8c3656ca009a15c7a2bb49f67ea5ecfc537397345863a6a1b7c091e27

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadClientTest.java (hunk 2 above)
Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatThreadClientTest.java:370]: I will update this test case to reflect the different behaviors

**Review by JianpingChen — APPROVED** [review]

**Review by juancamilor — APPROVED** [review]

**Pull request merged**